catalogue_api/api/src/main/scala/uk/ac/wellcome/platform/api/controllers/WorksController.scala

```diff
@@ -134,6 +139,24 @@ abstract class WorksController(apiConfig: ApiConfig,
     response.ok.json(ResultResponse(context = contextUri, result = result))
   }
 
+  /** Create a 302 Redirect to a new Work.
+    *
+    * Assumes the original URI requested was for a single work, i.e. a request
+    * of the form /works/{id}.
+    *
+    */
+  private def respondWithRedirect(originalUri: String,
+                                  work: IdentifiedRedirectedWork,
+                                  contextUri: String) =
+    response.found
+      .body("")
+      .location(
+        uri = originalUri.replaceAll(
+          s"/${work.canonicalId}",
+          s"/${work.redirect.canonicalId}"
+        )
```

Comment:
I wonder if it's worth anchoring this to ensure a single replacement of the last id?

Reply:
Seems sensible.

Reply:
Ah, but what do you anchor on? We don‘t want to match on the end of the string – then it doesn’t work for query parameters. I can’t think of a better way to do this that isn’t substantially more complicated. 😕